### PR TITLE
test and fix check blanks in var declarations

### PR DESCRIPTION
**Pull request by gburt**
I noticed that `var foo, _ = bar()` blanks weren't getting checked because `ast.GenDecl` statements (a subset of which are `var` declarations) weren't being handled/visited.  Fix that, and add back the analyzer tests for blank (and assert) flags.

To handle them, I refactored the existing assignment check.  It's completely unchanged except it now takes LHS and RHS args that we feed it from both the GenDecl and AssignStmt nodes' values.

### CODE REVIEW

File: errcheck/errcheck.go
Comment (gburt) [inline errcheck/errcheck.go:595]: by hiding whitespace when reviewing you can see this is just a move of the code and changing break -> return and stmt.Rhs -> rhs and stmt.Lhs -> lhs.

File: errcheck/testdata/src/blank/main.go
Comment (gburt) [inline errcheck/testdata/src/blank/main.go:34]: all tests passed except this check

**Comment by gburt** [comment]
btw thanks for this work, we're very glad to have it helping make our code better!

**Comment by kisielk** [comment]
Thanks very much for the thorough PR. Looks good to me but it seems there are some failing tests. Could you please take a look?

**Comment by gburt** [comment]
doh I'd just been running `go test -run TestAnalyzer/check_blanks` and hadn't noticed the other test failures

**Comment by gburt** [comment]
now we just need to make a "you visited AssignStmt but not GenDecl" linter :)

**Comment by kisielk** [comment]
Waiting for #221 to merge before merging this.

**Review by kisielk — APPROVED** [review]

**Pull request merged**
